web/frontend/src/components/navigation/components/map.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang='ts'>
+
+import { onMount, createEventDispatcher } from 'svelte';
+import { Map } from 'maplibre-gl';
+import { type LngLat } from '@/api/navigation';
+import { map } from '../stores';
+import { style } from '../style';
+import ObstacleLayer from './obstacle-layer.svelte';
+import Waypoints from './waypoints.svelte';
+import RobotMarker from './robot-marker.svelte';
+
+const dispatch = createEventDispatcher<{
+  drag: LngLat
+  dragstart: LngLat
+  dragend: LngLat
+}>();
+
+export let name: string;
+
+onMount(() => {
+  const mapInstance = new Map({
+    container: 'navigation-map',
+    style,
+    center: [0, 0],
+    zoom: 9,
+    pitch: 1,
+    antialias: true,
+    pitchWithRotate: false,
+  });
+
+  mapInstance.on('drag', () => dispatch('drag', mapInstance.getCenter()));
+  mapInstance.on('dragstart', () => dispatch('dragstart', mapInstance.getCenter()));
+  mapInstance.on('dragend', () => dispatch('dragend', mapInstance.getCenter()));
+
+  $map = mapInstance;
+});
+
+</script>
+
+<div
+  id='navigation-map'
+  class="mb-2 h-[550px] w-full"
```

Comment:
I am thinking about things here from a re-size perspective, i.e. minimizing the "screen" or moving the "screen" onto a larger computer monitor.

Does `w-full` resize  well when it comes to switching screen?